`Home` and `End` keys make lazygit crash when editing commit msg

**Describe the bug**
A clear and concise description of what the bug is.
When I edit the commit msg, I accidentally hit `Home` or `End`, and lazygit crashed.

**To Reproduce**
Steps to reproduce the behavior:
1. open lazygit in one git repo, stage a file using `Space`
2. `c` to edit commit msg
3. Hit `Home` or `End` key
4. See error

**Expected behavior**
A clear and concise description of what you expected to happen.

Lazy should just ignore the `Home` or `End` keys or better to hanle the `Home`/`End` which goes to the start/end of the commit msg

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Desktop (please complete the following information):**
 - OS: [e.g. Windows] `Manjaro-Linux`
 - Lazygit Version [e.g. v0.1.45] `0.25.1` installed from `go get`, and https://github.com/jesseduffield/lazygit/issues/1162
 - The last commit id if you built project from sources (run : ```git-rev parse HEAD```)

**Additional context**
Add any other context about the problem here.
```
>> lazygit
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/jesseduffield/gocui.(*View).writeRune(0xc0000ed8c0, 0x0, 0x0, 0xc000000000, 0xc000567778, 0x44a06c)
        /home/chz/.go/pkg/mod/github.com/jesseduffield/gocui@v0.3.1-0.20210208224444-2eecee85583d/edit.go:359 +0x6d8
github.com/jesseduffield/gocui.(*View).EditWrite(0xc0000ed8c0, 0x0)
        /home/chz/.go/pkg/mod/github.com/jesseduffield/gocui@v0.3.1-0.20210208224444-2eecee85583d/edit.go:68 +0x6b
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).commitMessageEditor(0xc000233600, 0xc0000ed8c0, 0x20fff1, 0x203000)
        /home/chz/.go/pkg/mod/github.com/jesseduffield/lazygit@v0.25.1/pkg/gui/editors.go:41 +0x16f
github.com/jesseduffield/gocui.EditorFunc.Edit(0xc00020f780, 0xc0000ed8c0, 0x38fff1, 0xae2100)
        /home/chz/.go/pkg/mod/github.com/jesseduffield/gocui@v0.3.1-0.20210208224444-2eecee85583d/edit.go:27 +0x4b
github.com/jesseduffield/gocui.(*Gui).onKey(0xc00026c6c0, 0xc0003803c0, 0xae2120, 0xc000260d01)
        /home/chz/.go/pkg/mod/github.com/jesseduffield/gocui@v0.3.1-0.20210208224444-2eecee85583d/gui.go:871 +0x344
github.com/jesseduffield/gocui.(*Gui).handleEvent(0xc00026c6c0, 0xc0003803c0, 0x0, 0x0)
        /home/chz/.go/pkg/mod/github.com/jesseduffield/gocui@v0.3.1-0.20210208224444-2eecee85583d/gui.go:551 +0x46
github.com/jesseduffield/gocui.(*Gui).MainLoop(0xc00026c6c0, 0xc000000004, 0xc00024da90)
        /home/chz/.go/pkg/mod/github.com/jesseduffield/gocui@v0.3.1-0.20210208224444-2eecee85583d/gui.go:499 +0x269
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).Run(0xc000233600, 0x0, 0x0)
        /home/chz/.go/pkg/mod/github.com/jesseduffield/lazygit@v0.25.1/pkg/gui/gui.go:504 +0x505
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).RunWithSubprocesses(0xc000233600, 0x0, 0xb0b1a0)
        /home/chz/.go/pkg/mod/github.com/jesseduffield/lazygit@v0.25.1/pkg/gui/gui.go:517 +0x108
github.com/jesseduffield/lazygit/pkg/app.(*App).Run(0xc000230a80, 0xc0000da750, 0x0)
        /home/chz/.go/pkg/mod/github.com/jesseduffield/lazygit@v0.25.1/pkg/app/app.go:240 +0x6b
main.main()
        /home/chz/.go/pkg/mod/github.com/jesseduffield/lazygit@v0.25.1/main.go:126 +0xb3b
```

The same happens when renaming a branch.
To reproduce
1. Open lazygit
2. Select a branch and rename it (Shift-R)
3. Confirm the message about local renaming by pressing `y`.
4. Hit `Home` or `End`.
5. See the error message.

```
panic: runtime error: index out of range [33] with length 33

goroutine 1 [running]:
github.com/jesseduffield/lazygit/vendor/github.com/jesseduffield/gocui.(*View).writeRune(0xc0002ac6e0, 0x21, 0x0, 0x0, 0xc0000f0080, 0xc000103718)
        github.com/jesseduffield/lazygit/vendor/github.com/jesseduffield/gocui/edit.go:359 +0x6d1
github.com/jesseduffield/lazygit/vendor/github.com/jesseduffield/gocui.(*View).EditWrite(0xc0002ac6e0, 0x0)
        github.com/jesseduffield/lazygit/vendor/github.com/jesseduffield/gocui/edit.go:68 +0x6b
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).defaultEditor(0xc000129600, 0xc0002ac6e0, 0xfff1, 0x1000000000000)
        github.com/jesseduffield/lazygit/pkg/gui/editors.go:72 +0x177
github.com/jesseduffield/lazygit/vendor/github.com/jesseduffield/gocui.EditorFunc.Edit(0xc000183a40, 0xc0002ac6e0, 0xefff1, 0x0)
        github.com/jesseduffield/lazygit/vendor/github.com/jesseduffield/gocui/edit.go:27 +0x4b
github.com/jesseduffield/lazygit/vendor/github.com/jesseduffield/gocui.(*Gui).onKey(0xc000202120, 0xc0000ee8a0, 0xb5f5e0, 0xc00000e401)
        github.com/jesseduffield/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:871 +0x33a
github.com/jesseduffield/lazygit/vendor/github.com/jesseduffield/gocui.(*Gui).handleEvent(0xc000202120, 0xc0000ee8a0, 0x3, 0x0)
        github.com/jesseduffield/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:551 +0x46
github.com/jesseduffield/lazygit/vendor/github.com/jesseduffield/gocui.(*Gui).MainLoop(0xc000202120, 0xc000000004, 0xc0001c3a80)
        github.com/jesseduffield/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:499 +0x3fb
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).Run(0xc000129600, 0x0, 0x0)
        github.com/jesseduffield/lazygit/pkg/gui/gui.go:504 +0x51a
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).RunWithSubprocesses(0xc000129600, 0x0, 0xb8b93e)
        github.com/jesseduffield/lazygit/pkg/gui/gui.go:517 +0x104
github.com/jesseduffield/lazygit/pkg/app.(*App).Run(0xc00016f9d0, 0xc00014a5a0, 0x0)
        github.com/jesseduffield/lazygit/pkg/app/app.go:240 +0x6b
main.main()
        github.com/jesseduffield/lazygit/main.go:126 +0xb2d
```

--- a/pkg/gui/editors.go
+++ b/pkg/gui/editors.go
@@ -1,6 +1,8 @@
 package gui
 
 import (
+	"unicode"
+
 	"github.com/jesseduffield/gocui"
 )
 
@@ -37,7 +39,7 @@ func (gui *Gui) commitMessageEditor(v *gocui.View, key gocui.Key, ch rune, mod g
 		v.EditGotoToStartOfLine()
 	case key == gocui.KeyCtrlE:
 		v.EditGotoToEndOfLine()
-	default:
+	case unicode.IsPrint(ch):
 		v.EditWrite(ch)
 	}
 
@@ -68,7 +70,7 @@ func (gui *Gui) defaultEditor(v *gocui.View, key gocui.Key, ch rune, mod gocui.M
 		v.EditGotoToStartOfLine()
 	case key == gocui.KeyCtrlE:
 		v.EditGotoToEndOfLine()
-	default:
+	case unicode.IsPrint(ch):
 		v.EditWrite(ch)
 	}
 

